From a73387efb5e605e48fbdfe2045534d7f6e3db6ba Mon Sep 17 00:00:00 2001
From: Chris Mi <chrism@mellanox.com>
Date: Thu, 10 May 2018 15:53:47 +0900
Subject: [PATCH 3/3] hash table for mlx5e_tc_flow is change in 4.17.0-rc3+

static struct rhashtable *get_tc_ht(struct mlx5e_priv *priv)
{
        struct mlx5_eswitch *esw = priv->mdev->priv.eswitch;
        struct mlx5e_rep_priv *uplink_rpriv;

        if (MLX5_VPORT_MANAGER(priv->mdev) && esw->mode == SRIOV_OFFLOADS) {
                uplink_rpriv = mlx5_eswitch_get_uplink_priv(esw, REP_ETH);
                return &uplink_rpriv->tc_ht;
        } else
                return &priv->fs.tc.ht;
}
---
 net.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/net.c b/net.c
index cbe652c..efc8af3 100644
--- a/net.c
+++ b/net.c
@@ -1954,11 +1954,11 @@ void show_mlx(ulong net_addr)
 
 	ulong tc = fs + MEMBER_OFFSET("mlx5e_flow_steering", "tc");
 	fprintf(fp, "mlx5e_tc_table  %lx\n", tc);
-	ulong ht = read_pointer2(tc, "mlx5e_tc_table", "ht");
-	fprintf(fp, "hash %lx -s mlx5e_tc_flow -m node\n", ht);
+/* 	ulong ht = read_pointer2(tc, "mlx5e_tc_table", "ht"); */
+	/* for old kernel before 4.17.0-rc3+ */
+/* 	fprintf(fp, "hash %lx -s mlx5e_tc_flow -m node\n", ht); */
 	/* mlx5e_tc_flow.cookie the address of cls_fl_filter */
 
-	// for old code
 	ulong  ppriv = read_pointer2(mlx5e_priv, "mlx5e_priv", "ppriv");
 /* 	fprintf(fp, "mlx5_eswitch_rep  %lx\n", ppriv); */
 
@@ -1966,10 +1966,15 @@ void show_mlx(ulong net_addr)
 	if (ppriv != 0) {
 		fprintf(fp, "mlx5e_rep_priv  %lx\n", ppriv);
 
-		ulong  rep = read_pointer2(ppriv, "mlx5e_rep_priv", "rep");
+		ulong rep = read_pointer2(ppriv, "mlx5e_rep_priv", "rep");
 		fprintf(fp, "mlx5_eswitch_rep  %lx\n", rep);
 		fprintf(fp, "mlx5_eswitch_rep  %lx\n", rep + STRUCT_SIZE("mlx5_eswitch_rep"));
 		fprintf(fp, "mlx5_eswitch_rep  %lx\n", rep + STRUCT_SIZE("mlx5_eswitch_rep") * 2);
+
+		ulong mlx5e_rep_priv = read_pointer2(rep, "mlx5_eswitch_rep_if", "priv");
+		fprintf(fp, "mlx5e_rep_priv  %lx\n", mlx5e_rep_priv);
+		ulong ht = read_pointer2(mlx5e_rep_priv, "mlx5e_rep_priv", "tc_ht");
+		fprintf(fp, "hash %lx -s mlx5e_tc_flow -m node\n", ht);
 	}
 
 	ulong mdev = read_pointer2(mlx5e_priv, "mlx5e_priv", "mdev");
-- 
2.14.2

